File: 1975-maximum-matrix-sum/1975-maximum-matrix-sum.cpp
class Solution {
public:
    long long maxMatrixSum(vector<vector<int>>& matrix) {
        int n = matrix.size(), m = matrix[0].size();
        long long sum = 0;
        int neg = 0;
        int minEl = INT_MAX;
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                sum += abs(matrix[i][j]);
                neg += matrix[i][j] < 0 ? 1 : 0;
                minEl = min(minEl, abs(matrix[i][j]));
            }
        }
        if (neg % 2 == 0) {
            return sum;
        }
        return sum - 2 * minEl;
    }
};

File: 1975-maximum-matrix-sum/1975-maximum-matrix-sum.py
class Solution:
    def maxMatrixSum(self, matrix: List[List[int]]) -> int:
        n = len(matrix)
        m = len(matrix[0])
        total_sum = 0
        neg_count = 0
        min_element = float('inf')

        for i in range(n):
            for j in range(m):
                total_sum += abs(matrix[i][j])
                if matrix[i][j] < 0:
                    neg_count += 1
                min_element = min(min_element, abs(matrix[i][j]))

        if neg_count % 2 == 0:
            return total_sum
        return total_sum - 2 * min_element